### PR TITLE
[#3800] Rails 4.2 Prerequisites

### PR DESCRIPTION
**Pull request by garethrees**
Prerequisite of #3800

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -184,4 +184,5 @@ group :development do
   gem 'launchy', '~> 2.4.0'
   gem 'spring', '~> 1.7.0', '< 2.0.0'
   gem 'spring-commands-rspec', '~> 1.0.0'
+  gem 'web-console', '~> 2.3.0', '< 3.0.0'
```
Comment (garethrees) [inline Gemfile]: Commit message should reference http://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#web-console

File: Gemfile [hunk 2]
```diff
@@ -90,7 +90,7 @@ gem 'cancancan', '~> 1.12.0', '< 1.13.0'
 gem 'charlock_holmes', '~> 0.7.0'
 gem 'dalli', '~> 2.7.0'
 gem 'dynamic_form', '~> 1.1.0'
-gem 'exception_notification', '~> 4.0.0', '< 4.1.0'
+gem 'exception_notification', '~> 4.1.0', '< 4.1.2'
```
Comment (garethrees) [inline Gemfile]: Need to understand what the bug is here

File: Gemfile [hunk 3]
```diff
@@ -142,7 +142,7 @@ gem 'unicode', '~> 0.4.0'
 gem 'unidecoder', '~> 1.1.0'
 
 # mime-types 3.0.0 requires Ruby 2.0.0, and _something_ is trying to update it
-gem 'mime-types', '< 3.0.0'
+gem 'mime-types', '< 3.0.0', require: false
```
Comment (garethrees) [inline Gemfile]: Commit message doesn't explain _why_ we've done this.

File: Gemfile (hunk 1 above)
Comment (garethrees) [inline Gemfile]: Added in fa52e73

File: Gemfile (hunk 3 above)
Comment (garethrees) [inline Gemfile]: Added in 5ade182

File: Gemfile (hunk 2 above)
Comment (garethrees) [inline Gemfile]: https://github.com/mysociety/alaveteli/commit/cb9b2a098fc7334d49d963092569d8ec42f6cf2b originally mentioned the existence of a bug.

Comment (lizconlan) [inline Gemfile]: I must have missed the fix in 4.1.1 (or - more likely - missed that 4.1.1 still had 1.9.3 support)

Comment (garethrees) [inline Gemfile]: Yep seems fine

**Pull request merged**